codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +404,237 @@ public TextHeaders clear() {
+            DefaultTextHeaders.super.set(name, commaSeparate(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.set(name, convertAndCommaSeparate(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.set(name, convertAndCommaSeparate(values));
+            return DefaultTextHeaders.this;
+        }
+
+        private TextHeaders addConverted(CharSequence name, CharSequence convertedValue) {
+            CharSequence currentValue = DefaultTextHeaders.super.get(name);
+            if (currentValue == null) {
+                DefaultTextHeaders.super.add(name, convertedValue);
+            } else {
+                DefaultTextHeaders.super.set(name, commaSeparate(currentValue, convertedValue));
+            }
+            return DefaultTextHeaders.this;
+        }
+
+        private CharSequence convertAndCommaSeparate(Object... values) {
+            StringBuilder singleValue = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0; i < end; i++) {
+                    convertAndQuote(singleValue, values[i]).append(", ");
+                }
+                convertAndQuote(singleValue, values[end]);
+            }
+            return singleValue.toString();
+        }
+
+        private CharSequence commaSeparate(CharSequence[] values) {
+            StringBuilder singleValue = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0; i < end; i++) {
+                    quote(singleValue, values[i]).append(", ");
+                }
+                quote(singleValue, values[end]);
+            }
+            return singleValue.toString();
+        }
+
+        private CharSequence convertAndCommaSeparate(Iterable<?> values) {
+            StringBuilder sb = new StringBuilder();
+            Iterator<?> iterator = values.iterator();
+            if (iterator.hasNext()) {
+                Object next = iterator.next();
+                while (iterator.hasNext()) {
+                    convertAndQuote(sb, next).append(", ");
+                    next = iterator.next();
+                }
+                convertAndQuote(sb, next);
+            }
+            return sb.toString();
+        }
+
+        private CharSequence commaSeparate(Iterable<? extends CharSequence> values) {
```

Comment:
The names of the interfaces are just for demonstration purposes.  I think they can be improved :)